### PR TITLE
Moved Cache class to core directory, as it's core functionality and n…

### PR DESCRIPTION
**Pull request by oisian**
There is no Dev branch is i had to PR to master.

Moved Cache class to core directory, as it's core functionality and not a model.

Added Caching to the settings, -> when we ask for a setting, if it is not already set, set it. else return the cached value. 

Added: remove key from Cache if we are setting a new value, as we will wan't that value next time there is a request for the setting.

### CODE REVIEW

**Comment by JavaPortals** [comment]
I find the PHP Storm comment at the top of the file unnecessary. Please remove and commit it.

**Comment by JavaPortals** [comment]
Cache.php#75 self::Remove($key) will throw a fatal error.

**Comment by oisian** [comment]
There you go, Didn't even notice them sorry

File: application/core/Cache.php [hunk 1]
```diff
@@ -35,36 +41,43 @@ public static function has($key) {
...
     /**
      * Get a value from cache and then remove from cache.
      * @param $key
      * @return bool
      */
-    public static function pull($key) {
+    public static function pull($key)
+    {
         $value = self::get($key);
         self::Remove($key);
```
Comment (JavaPortals) [inline application/core/Cache.php]: This will throw a FATAL PHP Error. Please fix it

Comment (oisian) [inline application/core/Cache.php]: No it won't why do you think that? 

It get's the key stores that in $value, Then removes the key from the cache and returns the value it got before removing it. 

i see no error

**Comment by oisian** [comment]
There you go

**Pull request merged**